### PR TITLE
Email edit mutation

### LINKED ISSUE #30: User Can Edit Email - BE
As a registered user, when I am on the profile edit page /user/profile/edit
I should see an editable pre-populated form with an email field.
When I edit the field and click save,
I should see a confirmation,
and be redirected to my profile page /user/profile
Where I see the updated email address.

- [ ] Add Update functionality for User email
- [ ] Update all GraphQL schemas and mutations

### PR DESCRIPTION
**Pull request by SMJ289**
## Type of change made:
- [ ] **Bugfix** (non-breaking change which fixes an issue)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] **Refactoring** (improves existing functionality)
- [x] **Testing** (tests existing code or improves existing tests)
- [ ] **Styling** (no functionality changed, just visual improvements)

## Description:
This adds a GraphQL mutation for editing User email and sets up testing environment.
Other contributors will need to create a 'crate_test' database locally (in the API subdirectory) and run the setup script:
In the psql console:
```
psql
CREATE DATABASE crate_test;
```
In the API directory:
```
NODE_ENV=test npm run setup:db
```
**Note: this only sets up testing for the API subdirectory**

## What functionality does this accomplish?

- Adds user email edit mutation
- Sets up Jest and SuperTest in the API subdirectory
- Added a test database
- Tests users and user by id queries

closes #30 

## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have fully tested my code
- [x] I have partially tested my code (please explain why): learning the new testing environment

### CODE REVIEW

File: code/api/src/modules/user/resolvers.js [hunk 1]
```diff
@@ -58,6 +58,20 @@ export async function login(parentValue, { email, password }) {
   }
 }
 
+// Update user
+export async function update(parentValue, { id, email }, { auth }) {
+  if (auth.user && auth.user.role === params.user.roles.user || params.user.roles.admin) {
```
Comment (rubyr) [inline code/api/src/modules/user/resolvers.js:63]: May have to change this around a bit in the future for security purposes, but for now it's fine I think

File: code/api/src/modules/user/mutations.js [hunk 2]
```diff
@@ -27,6 +27,23 @@ export const userSignup = {
...
+    id: {
+      name: 'id',
+      type: GraphQLInt
+    },
+
+    email: {
+      email: 'email',
+      type: GraphQLString
+    }
+  },
```
Comment (rubyr) [inline code/api/src/modules/user/mutations.js:43]: And I assume we'd add in the rest of the user info here, right

Comment (SMJ289) [reply code/api/src/modules/user/mutations.js:43]: Yep! After we add the columns to the database.

File: code/api/src/modules/user/resolvers.js (hunk 1 above)
Comment (SMJ289) [reply code/api/src/modules/user/resolvers.js:63]: Yeah, I am not sure how to verify the specifically authenticate a particular user.

File: code/api/src/modules/user/mutations.js [hunk 3]
```diff
@@ -27,6 +27,23 @@ export const userSignup = {
   resolve: create
 }
 
+// Crate update
+export const userUpdate = {
+  type: UserType,
+  args: {
+    id: {
```
Comment (J-Poulter) [inline code/api/src/modules/user/mutations.js:34]: for the userUpdate argument, where does the id come from?

**Pull request merged**